packages/react/src/components/Modal/ModalProvider.tsx

```diff
@@ -0,0 +1,159 @@
+import React, {
+  createContext,
+  createElement,
+  Fragment,
+  FunctionComponent,
+  PropsWithChildren,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState,
+} from "react";
+import {
+  DeleteConfirmationModal,
+  DeleteConfirmationModalProps,
+} from ":/components/Modal/DeleteConfirmationModal";
+import { randomString } from ":/utils";
+import { ModalProps, useModal } from ":/components/Modal/index";
+import {
+  ConfirmationModal,
+  ConfirmationModalProps,
+} from ":/components/Modal/ConfirmationModal";
+import { WithOptional } from ":/types";
+import {
+  MessageModal,
+  MessageModalProps,
+} from ":/components/Modal/MessageModal";
+
+export type Decision = string | null | undefined;
+export type DecisionModalProps = WithOptional<ModalProps, "size"> & {
+  onDecide: (decision?: Decision) => void;
+};
+// TODO: I don't really like this "& any", but it's the only way I found to make it work for MessageModal.
+type DecisionModalComponent = FunctionComponent<DecisionModalProps & any>;
```

Comment:
I would have made `messageType` optional for all modals (delete, confirm and message) with a default value as `INFO` for `MessageModal`.